dbt/seeds/location/docs.md

```diff
@@ -0,0 +1,23 @@
+# neighborhood_group
+
+{% docs seed_neighborhood_group %}
+A mapping of neighborhood codes to groups that are used for sales
+validation.
+
+### Nuance
+
+- The mapping table contains two rows that are used for versioning: `version` and
+  `updated_at`. The combination of these two columns allows us to reproduce the
+  state of the neighborhood groups at a specific moment in time. To do so, query
+  for all records where `updated_at` is below a certain date, group by `nbhd`,
+  and select the record with the highest value for `version` within the group.
+- Instructions for how to update the seed that generates this table:
+  - When **adding new neighborhoods**, append the rows to the CSV file with a
+    `version` of 1
+  - When **updating neighborhoods that already have a group**, append the rows
+    to the CSV file and increment the `version` by 1
+  - In the case of **all new rows**, set the `updated_at` column to the current
+    date
```

Comment:
This versioning scheme is similar to the type 2 slowly-changing dimension approach that dbt uses for [snapshots](https://docs.getdbt.com/docs/build/snapshots). I considered implementing versioning with snapshots instead -- the idea was something like an additional `location.neighborhood_group_history` table generated and updated by a snapshot that we could use to record the history of the groups whenever we change them -- but I didn't like the fact that the snapshots wouldn't be totally replicable from scratch in our DAG without the history of the table.

This all ties into deeper concerns about incremental tables that I'm hoping we can discuss during our infrastructure chat next week, but for now I went with the simplest option: Require the code author to manually implement the versioning any time they make a change. This is not ideal but I'm hoping that these groupings will change infrequently enough that it won't be a big deal (or that we'll have resolved our snapshot/incremental table design issue and we can refactor this by the time the next group of neighborhoods gets defined).